project-base/src/Shopsys/ShopBundle/Model/AdvancedSearch/ProductAdvancedSearchConfig.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+namespace Shopsys\ShopBundle\Model\AdvancedSearch;
+
+use Shopsys\FrameworkBundle\Model\AdvancedSearch\Filter\ProductAvailabilityFilter;
+use Shopsys\FrameworkBundle\Model\AdvancedSearch\Filter\ProductBrandFilter;
+use Shopsys\FrameworkBundle\Model\AdvancedSearch\Filter\ProductCalculatedSellingDeniedFilter;
+use Shopsys\FrameworkBundle\Model\AdvancedSearch\Filter\ProductCategoryFilter;
+use Shopsys\FrameworkBundle\Model\AdvancedSearch\Filter\ProductCatnumFilter;
+use Shopsys\FrameworkBundle\Model\AdvancedSearch\Filter\ProductFlagFilter;
+use Shopsys\FrameworkBundle\Model\AdvancedSearch\Filter\ProductNameFilter;
+use Shopsys\FrameworkBundle\Model\AdvancedSearch\Filter\ProductPartnoFilter;
+use Shopsys\FrameworkBundle\Model\AdvancedSearch\Filter\ProductStockFilter;
+use Shopsys\FrameworkBundle\Model\AdvancedSearch\ProductAdvancedSearchConfig as BaseProductAdvancedSearchConfig;
+
+class ProductAdvancedSearchConfig extends BaseProductAdvancedSearchConfig
+{
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\AdvancedSearch\Filter\ProductCatnumFilter $productCatnumFilter
+     * @param \Shopsys\FrameworkBundle\Model\AdvancedSearch\Filter\ProductNameFilter $productNameFilter
+     * @param \Shopsys\FrameworkBundle\Model\AdvancedSearch\Filter\ProductPartnoFilter $productPartnoFilter
+     * @param \Shopsys\FrameworkBundle\Model\AdvancedSearch\Filter\ProductStockFilter $productStockFilter
+     * @param \Shopsys\FrameworkBundle\Model\AdvancedSearch\Filter\ProductFlagFilter $productFlagFilter
+     * @param \Shopsys\FrameworkBundle\Model\AdvancedSearch\Filter\ProductCalculatedSellingDeniedFilter $productCalculatedSellingDeniedFilter
+     * @param \Shopsys\FrameworkBundle\Model\AdvancedSearch\Filter\ProductAvailabilityFilter $productAvailabilityFilter
+     * @param \Shopsys\FrameworkBundle\Model\AdvancedSearch\Filter\ProductBrandFilter $productBrandFilter
+     * @param \Shopsys\FrameworkBundle\Model\AdvancedSearch\Filter\ProductCategoryFilter $productCategoryFilter
+     */
+    public function __construct(
+        ProductCatnumFilter $productCatnumFilter,
+        ProductNameFilter $productNameFilter,
+        ProductPartnoFilter $productPartnoFilter,
+        ProductStockFilter $productStockFilter,
+        ProductFlagFilter $productFlagFilter,
+        ProductCalculatedSellingDeniedFilter $productCalculatedSellingDeniedFilter,
+        ProductAvailabilityFilter $productAvailabilityFilter,
+        ProductBrandFilter $productBrandFilter,
+        ProductCategoryFilter $productCategoryFilter
+    ) {
+        parent::__construct(
+            $productCatnumFilter,
+            $productNameFilter,
+            $productPartnoFilter,
+            $productStockFilter,
+            $productFlagFilter,
+            $productCalculatedSellingDeniedFilter,
+            $productAvailabilityFilter,
+            $productBrandFilter
+        );
+
+        $this->registerFilter($productCategoryFilter);
```

Comment:
we can create issue for that...into some milestone?

Reply:
@boris-brtan @grossmannmartin This could have been done wqithout needing changes in project repositories by just adding a call (see [Symfony docs](https://symfony.com/doc/current/service_container/calls.html)) in framework's `service.yml`:
```yml
Shopsys\FrameworkBundle\Model\AdvancedSearch\AdvancedSearchConfig:
    calls:
        - method: registerFilter
          arguments: ['@Shopsys\FrameworkBundle\Model\AdvancedSearch\Filter\ProductCategoryFilter']
```

We should probably redo this similarly to `GridInlineEdits` - search for `shopsys.grid_inline_edit` to see what I mean.

Reply:
seems to work with 
```yaml
Shopsys\FrameworkBundle\Model\AdvancedSearch\ProductAdvancedSearchConfig:
    calls:
        - method: registerFilter
          arguments: ['@Shopsys\FrameworkBundle\Model\AdvancedSearch\Filter\ProductCategoryFilter']
```